### ISSUE TITLE
Docs: Suggested object storage configuration for AWS S3 is invalid

### ISSUE THREAD

**Issue by albgus** [grafana/mimir#4228]
The object storage configuration page suggests using an environment variable to configure AWS keys, but this does not work. The docs should be updated, or the feature should actually be implemented.

This configuration works:
```
common:
  storage:
    backend: s3
    s3:
      endpoint: s3.eu-north-1.amazonaws.com
      region: eu-north-1
      access_key_id: "[REDACTED]"
      secret_access_key: "[REDACTED]"
```
However this does not:
```
common:
  storage:
    backend: s3
    s3:
      endpoint: s3.eu-north-1.amazonaws.com
      region: eu-north-1
      secret_access_key: "${AWS_SECRET_ACCESS_KEY}" # This is a secret injected via an environment variable
      access_key_id: "${AWS_ACCESS_KEY_ID}" # This is a secret injected via an environment variable
```

Rather it seems to time out, generating the following logs before crashing
```
ts=2023-02-11T12:15:48.247642705Z caller=sanity_check.go:41 level=error msg="Unable to successfully connect to configured object storage" err="blocks storage: unable to successfully send a request to object storage: The AWS Access Key Id you provided does not exist in our records."
ts=2023-02-11T12:15:48.247885622Z caller=mimir.go:802 level=error msg="module failed" module=sanity-check err="invalid service state: Failed, expected: Running, failure: blocks storage: unable to successfully send a request to object storage: The AWS Access Key Id you provided does not exist in our records."
ts=2023-02-11T12:15:48.247937695Z caller=mimir.go:802 level=error msg="module failed" module=ingester-service err="failed to start ingester-service, because it depends on module memberlist-kv, which has failed: context canceled"
ts=2023-02-11T12:15:48.247975386Z caller=mimir.go:802 level=error msg="module failed" module=memberlist-kv err="failed to start memberlist-kv, because it depends on module sanity-check, which has failed: invalid service state: Failed, expected: Running, failure: invalid service state: Failed, expected: Running, failure: blocks storage: unable to successfully send a request to object storage: The AWS Access Key Id you provided does not exist in our records."
ts=2023-02-11T12:15:48.248007411Z caller=mimir.go:802 level=error msg="module failed" module=store-queryable err="failed to start store-queryable, because it depends on module memberlist-kv, which has failed: context canceled"
ts=2023-02-11T12:15:48.248051499Z caller=mimir.go:802 level=error msg="module failed" module=server err="failed to start server, because it depends on module sanity-check, which has failed: invalid service state: Failed, expected: Running, failure: invalid service state: Failed, expected: Running, failure: blocks storage: unable to successfully send a request to object storage: The AWS Access Key Id you provided does not exist in our records."
ts=2023-02-11T12:15:48.248088387Z caller=mimir.go:802 level=error msg="module failed" module=distributor-service err="failed to start distributor-service, because it depends on module server, which has failed: context canceled"
ts=2023-02-11T12:15:48.248121879Z caller=mimir.go:802 level=error msg="module failed" module=querier err="failed to start querier, because it depends on module server, which has failed: context canceled"
ts=2023-02-11T12:15:48.248156712Z caller=mimir.go:802 level=error msg="module failed" module=ring err="failed to start ring, because it depends on module memberlist-kv, which has failed: context canceled"
ts=2023-02-11T12:15:48.248206677Z caller=mimir.go:802 level=error msg="module failed" module=ruler err="failed to start ruler, because it depends on module distributor-service, which has failed: context canceled"
ts=2023-02-11T12:15:48.248251553Z caller=mimir.go:802 level=error msg="module failed" module=store-gateway err="failed to start store-gateway, because it depends on module memberlist-kv, which has failed: context canceled"
ts=2023-02-11T12:15:48.248272291Z caller=mimir.go:802 level=error msg="module failed" module=compactor err="failed to start compactor, because it depends on module sanity-check, which has failed: invalid service state: Failed, expected: Running, failure: invalid service state: Failed, expected: Running, failure: blocks storage: unable to successfully send a request to object storage: The AWS Access Key Id you provided does not exist in our records."
ts=2023-02-11T12:15:48.248290495Z caller=mimir.go:802 level=error msg="module failed" module=query-frontend err="failed to start query-frontend, because it depends on module memberlist-kv, which has failed: context canceled"
ts=2023-02-11T12:15:48.26325723Z caller=log.go:65 level=error msg="error running application" err="failed services\ngithub.com/grafana/mimir/pkg/mimir.(*Mimir).Run\n\t/__w/mimir/mimir/pkg/mimir/mimir.go:845\nmain.main\n\t/__w/mimir/mimir/cmd/mimir/main.go:212\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:250\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1594"
```

I have confirmed that the environment variables in indeed injected
```
$ k exec -it mimir-5f5997f847-jlg54 -- env
PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
HOSTNAME=mimir-5f5997f847-jlg54
AWS_ACCESS_KEY_ID=[REDACTED]
AWS_SECRET_ACCESS_KEY=[REDACTED]
[...]
```

**Comment by pracucci** [comment]
To use environment variables in the config, you have to enable the expansion as documented here:
https://grafana.com/docs/mimir/latest/reference-configuration-parameters/#use-environment-variables-in-the-configuration

However, you're right that in the page you're mentioning (`docs/sources/mimir/operators-guide/configure/configure-object-storage-backend.md`) we're not mentioning it, while we should.

@albgus Willing to open a PR to clarify the doc, please?

**Comment by osg-grafana** [comment]
Johannes agreed to grab this on in our 1:1.

**Comment by 56quarters** [comment]
Fixed via #7509
